windows/src/engine/keyman32/appint/appint.h

```diff
@@ -74,28 +74,97 @@ class AppActionQueue
 	AppActionQueue();
 	virtual void ResetQueue();
 	virtual BOOL QueueAction(int ItemType, DWORD dwData);
-  
+
   BOOL IsQueueEmpty() { return QueueSize == 0; }
   int GetQueueSize() { return QueueSize; }
 };
 
 class AppContext
 {
 private:
-	WCHAR CurContext[MAXCONTEXT];
+	WCHAR CurContext[MAXCONTEXT]; //!< CurContext[0] is furthest from the caret and is null terminated.
 	int pos;
 
 public:
 	AppContext();
+  /**
+   * @brief Copy "source" AppContext to this AppContext
+   *
+   * @param source AppContext to copy
+   */
   void CopyFrom(AppContext *source);
-	void Add(WCHAR ch);
+
+  /**
+   * @brief  Add a single code unit to the Current Context. Not necessarily a complete code point
+   *
+   * @param Code unit to add
+   */
+  void Add(WCHAR ch);
```

Comment:
I can see us rewriting Add to be `Add(DWORD ch)`, and `AddDeadkey(UINT deadkeyCode)`... For the future. That would make this much better encapsulated!